### ISSUE TITLE
Index() on relations does not create index operations in migration scripts

### ISSUE THREAD

**Issue by jgburet** [mikro-orm/mikro-orm#1518]
This issue follows our conversation on Slack https://mikroorm.slack.com/archives/CFSAF8DEX/p1614768309156000

**Describe the bug**
No indexes are created when `Index()` decorates a `ManyToOne` relation (for example).

**Stack trace**
```
...
```

**To Reproduce**
```typescript
@Entity({ tableName: 'authors' })
export class Author {
  ... // PK

  @Field()
  @Index() // will be present in migration script
  name!: string
}

@Entity({ tableName: 'books' })
export class Book {
  ... // PK

  @ManyToOne(() => Book)
  @Index() // will NOT be present in migration script
  author!: IdentifiedReference<Author>;
}
```

```bash
yarn mikro-orm migration:create
```

**Expected behavior**
There should be two indexes created (without counting PKs):
- authors_name_index (present)
- books_author_id_index (missing)

**Additional context**
You need to have `Index()` defined for `author` on both the relation property and the entity class.
So this would work:
```typescript
@Entity({ tableName: 'books' })
@Index({ properties: 'author' }) // <-- added here
export class Book {
  ... // PK

  @ManyToOne(() => Book)
  @Index() // WILL BE present in migration script
  author!: IdentifiedReference<Author>;
}
```

**Additional context 2**
```typescript
@Entity({ tableName: 'books' })
@Index({ properties: 'author' })
export class Book {
  @ManyToOne
   // <-- missing here
  author!: IdentifiedReference<Author>;
}
```
This won't work either

**Versions**
Tested and detected in both `MK/core,migration,postgres` 4.3.4 and 4.4.4

        "@mikro-orm/cli": "^4.3.4",
        "@mikro-orm/core": "^4.3.4",
        "@mikro-orm/postgresql": "^4.3.4",

        "typescript": "^4.1.3"

nodejs 14.15.4

**Comment by jgburet** [comment]
A small touch of story telling first.
Yesterday evening I created new Entities and flagged my Indexes twice: one on the Entity with the `properties` attribute and one on the relation itself. When I ran `migration:create`, the index was being created twice in the resulting script. So I kept only one `Index()` call and that resulted in the expected behavior: one index declaration in the script.

My assumption right now is that there is a problem while diffing existing tables with entity classes.
- One call to `Index()` is enough when `migration:create` encounters the Entity class for the first time;
- Two calls to `Index()` are required for relations to have one index declaration to show up in the script.
